common/component/kafka/consumer.go

```diff
@@ -45,9 +44,9 @@ func notifyRecover(consumer *consumer, message *sarama.ConsumerMessage, session
 			// If the retry policy got interrupted, it could mean that either
 			// the policy has reached its maximum number of attempts or the context has been cancelled.
 			// There is a weird edge case where the error returned is a 'context canceled' error but the session.Context is not done.
-			// This is a workaround to handle that edge case and reprocess the current message.
-			if err == context.Canceled && session.Context().Err() == nil {
```

Comment:
the adjusted fix is 2-fold:
- the `err == context.Canceled` check was a bit too stringent and if the error is wrapped it may not match and therefore the condition would NOT be met
- More importantly, the recovery strategy is focusing more on the intent: If the policy is focused on infinite retry (in case zero data loss is paramount), if the session.Context() is NOT done, we would re-enter the loop regardless of the error...